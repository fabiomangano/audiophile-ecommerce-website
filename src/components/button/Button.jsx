/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import style from './Button.module.scss';

const Button = ({ children, type, onClick, variant, size }) => {
  return (
    <button
      className={cn(style[variant], style[size])}
      type={type}
      onClick={onClick}
      variant={variant}
      size={size}
    >
      {children}
    </button>
  );
};

Button.defaultProps = {
  type: 'button',
  children: null,
  onClick: undefined,
  variant: 'primary',
  size: 'md',
};

Button.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary', 'outline', 'link']),
  children: PropTypes.node,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  onClick: PropTypes.func,
  size: PropTypes.oneOf(['md']),
};

export default Button;
