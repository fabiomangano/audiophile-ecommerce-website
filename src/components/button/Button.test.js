import React from 'react';
import { create } from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  it('should renders correctly', () => {
    const tree = create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should output a button', () => {
    const wrapper = mount(<Button>Title</Button>);
    expect(wrapper.find('button').length).toEqual(1);
  });

  it('should set correctly the type when provided', () => {
    const wrapper = mount(<Button type="submit">Title</Button>);
    expect(wrapper.getDOMNode().getAttribute('type')).toEqual('submit');
  });

  it('should call onClick function when the user click', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button onClick={onClick}>Title</Button>);
    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('Should default to variant="primary and size="md"', () => {
    const wrapper = mount(<Button>Title</Button>);
    expect(wrapper.find('button').hasClass('primary')).toEqual(true);
    expect(wrapper.find('button').hasClass('md')).toEqual(true);
  });

  it('Should apply variant class', () => {
    const wrapper = mount(<Button variant="outline">Title</Button>);
    expect(wrapper.find('button').hasClass('outline')).toEqual(true);
  });
});
