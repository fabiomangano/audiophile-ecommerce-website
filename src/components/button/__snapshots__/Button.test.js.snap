// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders correctly 1`] = `
<button
  className="primary md"
  size="md"
  type="button"
  variant="primary"
/>
`;
